plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -44,36 +55,70 @@ public class DeleteModelTransportAction extends HandledTransportAction<ActionReq
+                log.debug("Completed Get Model Request, id:{}", modelId);
+
+                if (r != null && r.isExists()) {
+                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                        MLModel mlModel = MLModel.parse(parser);
+                        MLModelState mlModelState = mlModel.getModelState();
+                        System.out.println(mlModelState);
+                        if (mlModelState.equals(MLModelState.LOADED)
+                            || mlModelState.equals(MLModelState.LOADING)
+                            || mlModelState.equals(MLModelState.UPLOADING)
+                            || mlModelState.equals(MLModelState.PARTIALLY_LOADED)) {
+                            actionListener
+                                .onFailure(
+                                    new Exception("Model cannot be deleted in loading or loaded state. Try unloading first and then delete")
+                                );
+                        } else {
+                            DeleteRequest deleteRequest = new DeleteRequest(ML_MODEL_INDEX, modelId);
+                            client.delete(deleteRequest, new ActionListener<DeleteResponse>() {
+                                @Override
+                                public void onResponse(DeleteResponse deleteResponse) {
+                                    deleteModelChunks(modelId, deleteResponse, actionListener);
+                                }
 
-                @Override
-                public void onFailure(Exception e) {
-                    log.error("Failed to delete model meta data for model: " + modelId, e);
-                    if (e instanceof ResourceNotFoundException) {
-                        deleteModelChunks(modelId, null, actionListener);
+                                @Override
+                                public void onFailure(Exception e) {
+                                    log.error("Failed to delete model meta data for model: " + modelId, e);
+                                    if (e instanceof ResourceNotFoundException) {
+                                        deleteModelChunks(modelId, null, actionListener);
+                                    }
+                                    actionListener.onFailure(e);
+                                }
+                            });
+                        }
+                    } catch (Exception e) {
+                        log.error("Failed to parse ml model" + r.getId(), e);
+                        actionListener.onFailure(e);
                     }
-                    actionListener.onFailure(e);
                 }
-            });
+            }, e -> { actionListener.onFailure(new MLResourceNotFoundException("Fail to find model")); }), () -> context.restore()));
```

Comment:
It's possible that user deleted model meta doc first, then failed to delete model chunk when run line 103. That's why we add line 110 which will re-delete model chunks even model meta doc not found. I think we should follow the same logic here.

Reply:
This is for the outer wrap when the get model request does not find a model. Line 110 corresponds to delete request. In this case, the delete request is not called yet. So, skipped deleting anything and just threw an exception.